File: Aula_Ricardo/Programação_orientada_a_objetos/python/Polimorfismo.py
class Animal:
    def som(self):
        raise NotImplementedError("")
    
class Vaca(Animal):
    def som(self):
        return "Auau"
    
class gato(Animal):
    def som(self):
        return "MUUU"
    
def emitir_som(animais):
    for animal in animais:
        print(animal.som())

animais = [Vaca(), gato()]
emitir_som(animais)

File: Aula_Ricardo/Programação_orientada_a_objetos/java/Polimorfismo.java
abstract class Animal {
    abstract String som();
}

class Vaca extends Animal {
    @Override
    String som() {
        return "Muu";
    }
}

class Gato extends Animal {
    @Override
    String som() {
        return "Miau";
    }
}

public class Main {
    public static void emitirSom(Animal[] animais) {
        for (Animal animal : animais) {
            System.out.println(animal.som());
        }
    }

    public static void main(String[] args) {
        Animal[] animais = {new Vaca(), new Gato()};

        emitirSom(animais);
    }
}
